node/builder_chain.go

```diff
@@ -239,7 +238,7 @@ func ConfigFullNode(c interface{}) Option {
 		// If the Eth JSON-RPC is enabled, enable storing events at the ChainStore.
 		// This is the case even if real-time and historic filtering are disabled,
 		// as it enables us to serve logs in eth_getTransactionReceipt.
-		If(cfg.Fevm.EnableEthRPC || cfg.Events.EnableActorEventsAPI, Override(StoreEventsKey, modules.EnableStoringEvents)),
+		If(cfg.Fevm.EnableEthRPC || cfg.Events.EnableActorEventsAPI || !cfg.ChainIndexer.DisableIndexer, Override(StoreEventsKey, modules.EnableStoringEvents)),
```

Comment:
Does this make sense? You have a check for `IsStoringEvents` in, so surely it's OK to enable indexer but not get events stored if you don't need them?

Back to the discussion @ https://github.com/filecoin-project/lotus/pull/12421/files#r1758128916 - imagine a scenario where we're indexing a lot more things, small things that help us do quicker look-ups than fishing around in badger. We wouldn't want to foist events on users that never need them just to get the benefits of this. So maybe we should be careful not to engineer a binary here.

Reply:
@rvagg As discussed, we want to either index everything or nothing. We have yet to see a concrete use case for wanting partial indexing. If someone comes up with it, we can always remove this but let's keep it here for now. We don't want to go down the path of partial indexing((makes the indexed state hard to reason about down the line)) unless there is user driven forcing function.

Reply:
@rvagg Actually, after dogfooding this and thinking through this, you have a point. This spooky action at a distance does not make sense and will confuse RPC providers about what is being indexed. Reverting this change.